misskey: updateScript fails to complete

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

Attempting to run the `passthru.updateScript` for `misskey` results in a failure. This is abit annoying, as any failure interrupts maintainer-wide update runs.

### Steps to reproduce

`nix-shell maintainers/scripts/update.nix --argstr package misskey`

### Expected behaviour

Running the updateScript should not fail.

### Screenshots

_No response_

### Relevant log output

```console
- misskey-2025.6.3: UPDATING ...
 - misskey-2025.6.3: ERROR

--- SHOWING ERROR LOG FOR misskey-2025.6.3 ----------------------

warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
this derivation will be built:
  /nix/store/3km3ghcpm2i9ixs71qginfqvsndznh6n-misskey-pnpm-deps.drv
building '/nix/store/3km3ghcpm2i9ixs71qginfqvsndznh6n-misskey-pnpm-deps.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/psmyqsf6p71v2g3xvlamisa4jx7453lz-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: installPhase
/build /build/source
/build/source
 WARN  using --force I sure hope you know what you are doing
Scope: all 11 workspace projects
 ERR_PNPM_LOCKFILE_CONFIG_MISMATCH  Cannot proceed with the frozen installation. The current "patchedDependencies" configuration doesn't match the value found in the lockfile

Update your lockfile using "pnpm install --no-frozen-lockfile"
error: builder for '/nix/store/3km3ghcpm2i9ixs71qginfqvsndznh6n-misskey-pnpm-deps.drv' failed with exit code 1;
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.6, NixOS, 25.11 (Xantusia), 25.11pre831064.6e987485eb2c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/puna/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/puna/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/puna/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/puna/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/3skx6kzlfq5c60bbyd181h7shvmqis7z-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/dm5smb3n6i76mpl3j29rrj0jh529zwcd-lix-2.91.3/share`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos, nixos-25.05-25.05, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@feathecutie @eljamm @ethancedwards8 @fricklerhandwerk @OPNA2608 @Prince213 @wegank

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/mi/misskey/package.nix
+++ b/pkgs/by-name/mi/misskey/package.nix
@@ -17,15 +17,19 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "misskey";
-  version = "2025.6.3";
+  version = "2025.7.0";
 
   src = fetchFromGitHub {
     owner = "misskey-dev";
     repo = "misskey";
     tag = finalAttrs.version;
-    hash = "sha256-6UZcIZlfcYcQgjR/jrNhsoLNQGml2tjK3LYLI0fdgMU=";
+    hash = "sha256-LtBggq60buNPnGPSbh+TcFODxCoqX+rFdX0P7dYMYI0=";
     fetchSubmodules = true;
   };
+
+  patches = [
+    ./pnpm-lock.yaml.patch
+  ];
 
   nativeBuildInputs = [
     nodejs
@@ -36,9 +40,14 @@ stdenv.mkDerivation (finalAttrs: {
 
   # https://nixos.org/manual/nixpkgs/unstable/#javascript-pnpm
   pnpmDeps = pnpm_9.fetchDeps {
-    inherit (finalAttrs) pname version src;
-    fetcherVersion = 1;
-    hash = "sha256-T8LwpEjeWNmkIo3Dn1BCFHBsTzA/Dt6/pk/NMtvT0N4=";
+    inherit (finalAttrs)
+      pname
+      version
+      src
+      patches
+      ;
+    fetcherVersion = 2;
+    hash = "sha256-5yuM56sLDSo4M5PDl3gUZOdSexW1YjfYBR3BJMqNHzU=";
   };
 
   buildPhase = ''

--- a/pkgs/by-name/mi/misskey/pnpm-lock.yaml.patch
+++ b/pkgs/by-name/mi/misskey/pnpm-lock.yaml.patch
@@ -1,0 +1,270 @@
+--- a/pnpm-lock.yaml
++++ b/pnpm-lock.yaml
+@@ -11,7 +11,7 @@
+ 
+ patchedDependencies:
+   typeorm:
+-    hash: 2677b97a423e157945c154e64183d3ae2eb44dfa9cb0e5ce731a7612f507bb56
++    hash: i7ls76affxbomopkwkccq5jvsu
+     path: patches/typeorm.patch
+ 
+ importers:
+@@ -51,6 +51,10 @@
+       typescript:
+         specifier: 5.8.3
+         version: 5.8.3
++    optionalDependencies:
++      '@tensorflow/tfjs-core':
++        specifier: 4.22.0
++        version: 4.22.0(encoding@0.1.13)
+     devDependencies:
+       '@misskey-dev/eslint-plugin':
+         specifier: 2.1.0
+@@ -85,10 +89,6 @@
+       start-server-and-test:
+         specifier: 2.0.12
+         version: 2.0.12
+-    optionalDependencies:
+-      '@tensorflow/tfjs-core':
+-        specifier: 4.22.0
+-        version: 4.22.0(encoding@0.1.13)
+ 
+   packages/backend:
+     dependencies:
+@@ -427,7 +427,7 @@
+         version: 4.2.0
+       typeorm:
+         specifier: 0.3.24
+-        version: 0.3.24(patch_hash=2677b97a423e157945c154e64183d3ae2eb44dfa9cb0e5ce731a7612f507bb56)(ioredis@5.6.1)(pg@8.16.0)(reflect-metadata@0.2.2)
++        version: 0.3.24(patch_hash=i7ls76affxbomopkwkccq5jvsu)(ioredis@5.6.1)(pg@8.16.0)(reflect-metadata@0.2.2)
+       typescript:
+         specifier: 5.8.3
+         version: 5.8.3
+@@ -446,6 +446,94 @@
+       xev:
+         specifier: 3.0.2
+         version: 3.0.2
++    optionalDependencies:
++      '@swc/core-android-arm64':
++        specifier: 1.3.11
++        version: 1.3.11
++      '@swc/core-darwin-arm64':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-darwin-x64':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-freebsd-x64':
++        specifier: 1.3.11
++        version: 1.3.11
++      '@swc/core-linux-arm-gnueabihf':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-linux-arm64-gnu':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-linux-arm64-musl':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-linux-x64-gnu':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-linux-x64-musl':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-win32-arm64-msvc':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-win32-ia32-msvc':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@swc/core-win32-x64-msvc':
++        specifier: 1.12.0
++        version: 1.12.0
++      '@tensorflow/tfjs':
++        specifier: 4.22.0
++        version: 4.22.0(encoding@0.1.13)(seedrandom@3.0.5)
++      '@tensorflow/tfjs-node':
++        specifier: 4.22.0
++        version: 4.22.0(encoding@0.1.13)(seedrandom@3.0.5)
++      bufferutil:
++        specifier: 4.0.9
++        version: 4.0.9
++      slacc-android-arm-eabi:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-android-arm64:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-darwin-arm64:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-darwin-universal:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-darwin-x64:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-freebsd-x64:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-linux-arm-gnueabihf:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-linux-arm64-gnu:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-linux-arm64-musl:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-linux-x64-gnu:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-linux-x64-musl:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-win32-arm64-msvc:
++        specifier: 0.0.10
++        version: 0.0.10
++      slacc-win32-x64-msvc:
++        specifier: 0.0.10
++        version: 0.0.10
++      utf-8-validate:
++        specifier: 6.0.5
++        version: 6.0.5
+     devDependencies:
+       '@jest/globals':
+         specifier: 29.7.0
+@@ -612,94 +700,6 @@
+       supertest:
+         specifier: 7.1.1
+         version: 7.1.1
+-    optionalDependencies:
+-      '@swc/core-android-arm64':
+-        specifier: 1.3.11
+-        version: 1.3.11
+-      '@swc/core-darwin-arm64':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-darwin-x64':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-freebsd-x64':
+-        specifier: 1.3.11
+-        version: 1.3.11
+-      '@swc/core-linux-arm-gnueabihf':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-linux-arm64-gnu':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-linux-arm64-musl':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-linux-x64-gnu':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-linux-x64-musl':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-win32-arm64-msvc':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-win32-ia32-msvc':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@swc/core-win32-x64-msvc':
+-        specifier: 1.12.0
+-        version: 1.12.0
+-      '@tensorflow/tfjs':
+-        specifier: 4.22.0
+-        version: 4.22.0(encoding@0.1.13)(seedrandom@3.0.5)
+-      '@tensorflow/tfjs-node':
+-        specifier: 4.22.0
+-        version: 4.22.0(encoding@0.1.13)(seedrandom@3.0.5)
+-      bufferutil:
+-        specifier: 4.0.9
+-        version: 4.0.9
+-      slacc-android-arm-eabi:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-android-arm64:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-darwin-arm64:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-darwin-universal:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-darwin-x64:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-freebsd-x64:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-linux-arm-gnueabihf:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-linux-arm64-gnu:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-linux-arm64-musl:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-linux-x64-gnu:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-linux-x64-musl:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-win32-arm64-msvc:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      slacc-win32-x64-msvc:
+-        specifier: 0.0.10
+-        version: 0.0.10
+-      utf-8-validate:
+-        specifier: 6.0.5
+-        version: 6.0.5
+ 
+   packages/frontend:
+     dependencies:
+@@ -11221,8 +11221,8 @@
+   vue-component-type-helpers@2.2.12:
+     resolution: {integrity: sha512-YbGqHZ5/eW4SnkPNR44mKVc6ZKQoRs/Rux1sxC6rdwXb4qpbOSYfDr9DsTHolOTGmIKgM9j141mZbBeg05R1pw==}
+ 
+-  vue-component-type-helpers@3.0.1:
+-    resolution: {integrity: sha512-j23mCB5iEbGsyIhnVdXdWUOg+UdwmVxpKnYYf2j+4ppCt5VSFXKjwu9YFt0QYxUaf5G99PuHsVfRScjHCRSsGQ==}
++  vue-component-type-helpers@3.0.3:
++    resolution: {integrity: sha512-koiBu7lO8e6w/UlbZAAIW11qcFQocYIl7Nh/SVwGZ804ej5KrncU32bRxi2zfU2Kyf6HWuk1CeeVP2rhIL+vyQ==}
+ 
+   vue-demi@0.14.7:
+     resolution: {integrity: sha512-EOG8KXDQNwkJILkx/gPcoL/7vH+hORoBaKgGe+6W7VFMvCYJfmF2dGbvgDroVnI8LU7/kTu8mbjRZGBU1z9NTA==}
+@@ -14955,7 +14955,7 @@
+       ts-dedent: 2.2.0
+       type-fest: 2.19.0
+       vue: 3.5.17(typescript@5.8.3)
+-      vue-component-type-helpers: 3.0.1
++      vue-component-type-helpers: 3.0.3
+ 
+   '@stylistic/eslint-plugin@2.13.0(eslint@9.31.0)(typescript@5.8.3)':
+     dependencies:
+@@ -23034,7 +23034,7 @@
+ 
+   typedarray@0.0.6: {}
+ 
+-  typeorm@0.3.24(patch_hash=2677b97a423e157945c154e64183d3ae2eb44dfa9cb0e5ce731a7612f507bb56)(ioredis@5.6.1)(pg@8.16.0)(reflect-metadata@0.2.2):
++  typeorm@0.3.24(patch_hash=i7ls76affxbomopkwkccq5jvsu)(ioredis@5.6.1)(pg@8.16.0)(reflect-metadata@0.2.2):
+     dependencies:
+       '@sqltools/formatter': 1.2.5
+       ansis: 3.17.0
+@@ -23371,7 +23371,7 @@
+ 
+   vue-component-type-helpers@2.2.12: {}
+ 
+-  vue-component-type-helpers@3.0.1: {}
++  vue-component-type-helpers@3.0.3: {}
+ 
+   vue-demi@0.14.7(vue@3.5.17(typescript@5.8.3)):
+     dependencies:

